SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/DynamicSubobjectsTest/DynamicSubobjectsTest.cpp

```diff
@@ -52,198 +64,258 @@ void ADynamicSubobjectsTest::PrepareTest()
-			[this](float DeltaTime) {
-				ASpatialFunctionalTestFlowController* FlowController = GetLocalFlowController();
-				APlayerController* PlayerController = Cast<APlayerController>(FlowController->GetOwner());
-				ATestMovementCharacter* PlayerCharacter = Cast<ATestMovementCharacter>(PlayerController->GetPawn());
-				if (IsValid(PlayerController))
-				{
-					if (PlayerCharacter == PlayerController->AcknowledgedPawn)
+		// When in native, we need to wait for a while here - so the engine can update relevancy
+		const bool bIsSpatial = Cast<USpatialNetDriver>(GetNetDriver()) != nullptr;
+		if (!bIsSpatial)
+		{
+			AddStep(
+				TEXT("DynamicSubobjectsTestNativeWaitABit"), FWorkerDefinition::Server(1), nullptr,
+				[this]() {
+					StepTimer = 0.f;
+				},
+				[this](float DeltaTime) {
+					StepTimer += DeltaTime;
+					if (StepTimer > 7.5f)
 					{
 						FinishStep();
 					}
+				});
+		}
+
+		// Step 8 - Server increases ADynamicSubObjectTestActor's TestIntProperty to enable checking if the client is out of interest later.
+		AddStep(TEXT("DynamicSubobjectsTestServerIncreasesIntValue"), FWorkerDefinition::Server(1), nullptr, [this, i]() {
+			TestActor->TestIntProperty = i;
+			FinishStep();
+		});
+
+		// Step 9 - Client 1 checks it can no longer see the ADynamicSubObjectTestActor by waiting for 0.5s and checking TestIntProperty
+		// hasn't updated
+		AddStep(
+			TEXT("DynamicSubobjectsTestClientCheckIntValueIncreased"), FWorkerDefinition::Client(1), nullptr,
+			[this]() {
+				StepTimer = 0.f;
+			},
+			[this, i](float DeltaTime) {
+				RequireNotEqual_Int(TestActor->TestIntProperty, i, TEXT("Check TestIntProperty didn't get replicated"));
+				StepTimer += DeltaTime;
+				if (StepTimer >= 0.5f)
+				{
+					FinishStep();
 				}
 			},
 			StepTimeLimit);
-	}
 
-	for (int i = 0; i < DynamicComponentsPerClass + 1; ++i)
-	{
-		{ // Step 3 - Server moves the TestMovementCharacter of Client 1 to a remote location, so that it does not see the
-		  // AReplicatedGASTestActor.
-			AddStep(TEXT("DynamicSubobjectsTestServerMoveClient1"), FWorkerDefinition::Server(1), nullptr, [this, i]() {
-				if (ClientOneSpawnedPawn->SetActorLocation(CharacterRemoteLocation))
+		if (bLastStepLoop)
+		{
+			// Step 9.1 - Server removes the component for secondary test case
+			AddStep(TEXT("DynamicSubobjectsTestServerDestroyActorComponent"), FWorkerDefinition::Server(1), nullptr, [this]() {
+				TArray<USceneComponent*> AllSceneComps;
+				TestActor->GetComponents<USceneComponent>(AllSceneComps);
+				AssertEqual_Int(AllSceneComps.Num(), InitialNumComponents + 1,
+								TEXT("ADynamicSubObjectTestActor should have 1 more than the initial number of components"));
+
+				// Delete the component with the right name
+				for (USceneComponent* SceneComponent : AllSceneComps)
 				{
-					if (ClientOneSpawnedPawn->GetActorLocation().Equals(CharacterRemoteLocation, 1.0f))
+					if (SceneComponent->GetName() == TEXT("ToRemoveComponent"))
```

Comment:
Small nit: could also make const string somewhere to not have to repeat the actual string, but it's just 2 places, so nbd